File: Delete a Node in Single Linked List[30-August-2023]/Solution.cpp
//   >> CPP CODE <<

Node* deleteNode(Node *head,int x)
{
    if(x==1) return head->next;
    
    Node* prev=NULL;
    Node* ptr=head;
    
    while(x>1)
    {
        prev=ptr;
        ptr=ptr->next;
        x-=1;
    }
    
    prev->next=ptr->next;
    
    return head;
}

File: Delete a Node in Single Linked List[30-August-2023]/Solution.java
//   >> JAVA CODE <<

class GfG
{
    Node deleteNode(Node head, int x)
    {
	    if(x==1) return head.next;
	    
	    Node ptr=head;
	    Node prev=null;
	    
	    while(x>1)
	    {
	        prev=ptr;
	        ptr=ptr.next;
	        x-=1;
	    }
	    
	    prev.next=ptr.next;
	    
	        return head;
    }
}